ctapipe/image/extractor.py

```diff
@@ -743,38 +743,27 @@ def _apply_first_pass(
         # event.dl0.tel[tel_id].waveform object has shape (N_pixels, N_samples)
 
         # For each pixel, we slide a 3-samples window through the
-        # waveform without touching the extremes (so later we can increase it
-        # to 5), summing each time the ADC counts contained within it.
+        # waveform summing each time the ADC counts contained within it.
 
-        # 'width' could be configurable in a generalized version
-        # Right now this image extractor is optimized for LSTCam and NectarCam
         width = self.peak_finding_window_width.tel[telid]
         sums = convolve1d(waveforms, np.ones(width), axis=1, mode="nearest")
```

Comment:
did you try this with `method='mirror'`, which I think is more similar to what is done in CTA-MARS?  That might fix the slight differences, if they are due to edge cases (literally pulses at the edge of the readout window).  Though it's hard to see how that would be intensity dependent.

Reply:
looking at the previous implementation in 0.8, I think pulses that were in the edge bins were simply ignored, however

Reply:
yes this was indeed the case, because in CTAMARS the 3-samples peak search window never reaches the edges, but it stops before the last sample(s) of the readout window, allowing to increase the window as 1+3+1 (do I remember correctly @moralejo ?)

Reply:
So change to `peak_index = argmax([:, 1:-1], axis=1)` to avoid the edges?

Reply:
> yes this was indeed the case, because in CTAMARS the 3-samples peak search window never reaches the edges, but it stops before the last sample(s) of the readout window, allowing to increase the window as 1+3+1 (do I remember correctly @moralejo ?)

Yes, that is how it is done.